nixos/modules/services/web-apps/netbox.nix

```
@@ -0,0 +1,187 @@
+              'PORT': 6379,
+              # Comment out `HOST` and `PORT` lines and uncomment the following if using Redis Sentinel
+              # 'SENTINELS': [('mysentinel.redis.example.com', 6379)],
+              # 'SENTINEL_SERVICE': 'netbox',
+              'DATABASE': 1,
+              'SSL': False,
+          }
+      }
+
+      with open("${cfg.secretKeyFile}", "r") as file:
+          SECRET_KEY = file.readline()
+
+      ${cfg.extraConfig}
+    '';
+  };
+  pkg = (cfg.package.overrideAttrs (old: {
+    paths = old.paths ++ [
+      (pkgs.runCommand "netbox-config" {} ''
+        install -d $out/opt/netbox/netbox/netbox/
+        ln -s ${configFile} $out/opt/netbox/netbox/netbox/configuration.py
+        '')
+    ];
+  })).override {
+    plugins = cfg.plugins;
+  };
+
+in {
+  options.services.netbox = {
+    enable = mkEnableOption "Netbox";
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "[::1]";
+      description = ''
+        Address the server will listen on.
+      '';
+    };
+
+    port = mkOption {
+      type = types.port;
+      default = 8001;
+      description = ''
+        Port the server will listen on.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.netbox;
+      defaultText = literalExpression "pkgs.netbox";
+      description = ''
+        Netbox package to use.
+      '';
+    };
+
+    plugins = mkOption {
+      type = types.listOf types.package;
+      default = [];
+      description = ''
+        Python-Packages to install as Plugins
+      '';
+    };
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/netbox";
+      description = "Storage path of netbox.";
+    };
+
+    secretKeyFile = mkOption {
+      type = types.path;
+      description = ''
+        A file containing the secret key
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Additional lines of configuration appended to the <literal>configuration.py</literal>
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.redis.enable = true;
+
+    services.postgresql = {
+      enable = true;
+      ensureDatabases = [ "netbox" ];
+      ensureUsers = [
+        {
+          name = "netbox";
+          ensurePermissions = {
+            "DATABASE netbox" = "ALL PRIVILEGES";
+          };
+        }
+      ];
+    };
+
+    systemd.services.netbox = {
+      description = "NetBox WSGI Service";
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      preStart = ''
+        ${pkg}/bin/python ${pkg}/opt/netbox/netbox/manage.py migrate
+        ${pkg}/bin/python ${pkg}/opt/netbox/netbox/manage.py trace_paths --no-input
+        ${pkg}/bin/python ${pkg}/opt/netbox/netbox/manage.py collectstatic --no-input
+        ${pkg}/bin/python ${pkg}/opt/netbox/netbox/manage.py remove_stale_contenttypes --no-input
+      '';
+
```

Then use `pkg.pythonPath` to pass the python dependencies into in the systemd unit's environment.

```suggestion
      environment = {
        PYTHONPATH = pkg.pythonPath; # this is the result of `makePythonPath` put into passthru
      };
```